### PR TITLE
OPS-1536: Add python3 support to xqwatcher

### PR DESCRIPTION
**Pull request by maxrothman**
@edx/devops review

### CODE REVIEW

**Comment by maxrothman** [comment]
@edx/devops looking for directional review and input on testing plan on 4dbc15c

**Comment by maxrothman** [comment]
@edx/devops this is ready for review
@jcdyer I can now build AMIs. How should we go about testing?

File: playbooks/roles/aws/tasks/main.yml [hunk 1]
```diff
@@ -87,13 +87,27 @@
...
+- name: Update the ssh motd on Ubuntu
+  file:
+    path: "{{ item.0 }}"
+    mode: "0644"
+  when: >
+    vagrant_home_dir.stat.exists == false and
+    ansible_distribution in common_debian_variants and
+    item.1.stat.exists
+  with_together:
+    - - "/etc/update-motd.d/10-help-text"
```
Comment (feanil) [inline playbooks/roles/aws/tasks/main.yml]: Instead of repeating this twice, let's move this list to a var in `defaults/main.yml`

File: playbooks/roles/common/tasks/main.yml [hunk 2]
```diff
@@ -34,24 +34,32 @@
     path: /home/vagrant
   register: vagrant_home_dir
 
+# Ensure that we get a current version of Git
```
Comment (feanil) [inline playbooks/roles/common/tasks/main.yml]: Why was this moved above adding the apt repository?

Comment (feanil) [inline playbooks/roles/common/tasks/main.yml]: NM, misread this.

File: playbooks/roles/supervisor/tasks/main.yml [hunk 3]
```diff
@@ -121,6 +121,19 @@
     dest: "/etc/init/{{ supervisor_service }}.conf"
     owner: root
     group: root
+  when: ansible_distribution_major_version < 16
+  tags:
+    - install
+    - install:base
+
+# NB: with systemd, pre_supervisor is a pre-task for supervisor, not a separate service
+- name: Create supervisor systemd job
```
Comment (feanil) [inline playbooks/roles/supervisor/tasks/main.yml]: NIT: Can you move this task above or below the two installs of 'init.d' scripts.  That way visually all the old tasks and the new tasks are separated and not intermingled as much.

**Comment by jcdyer** [comment]
@maxrothman Let's set up a box with the new branch of XQueueWatcher and a python3 queue and point an edx-platform sandbox at it.  I'll create a course on the sandbox that has a python3 coderesponse problem.

**Comment by maxrothman** [comment]
@jcdyer I think the easiest thing would be to grab stage for an hour or two.

**Comment by maxrothman** [comment]
@feanil addressed comments

**Comment by maxrothman** [comment]
To help out with review, here's what a new templated systemd file looks like:

```
[Unit]
Description=supervisord - Supervisor process control system
Documentation=http://supervisord.org
After=network.target


[Service]
# Run pre_supervisor
ExecStartPre=/edx/app/supervisor/venvs/supervisor/bin/python /edx/app/supervisor/pre_supervisor_checks.py \
  --hipchat-api-key <key> --hipchat-room release \
  --available=/edx/app/supervisor/conf.available.d --enabled=/edx/app/supervisor/conf.d

Type=forking
User=www-data
TimeoutStartSec=432000

ExecStart=/edx/app/supervisor/venvs/supervisor/bin/supervisord -n --configuration /edx/app/supervisor/supervisord.conf
ExecReload=/edx/app/supervisor/venvs/supervisor/bin/supervisorctl reload
ExecStop=/edx/app/supervisor/venvs/supervisor/bin/supervisorctl shutdown


[Install]
WantedBy=multi-user.target
```

**Comment by maxrothman** [comment]
@edx/devops this is finalized. Rereview?

File: playbooks/roles/common/tasks/main.yml [hunk 4]
```diff
@@ -34,24 +34,32 @@
...
+# GitHub requires version 1.7.10 or later
+# https://help.github.com/articles/https-cloning-errors
```
Comment (fredsmith) [inline playbooks/roles/common/tasks/main.yml:39]: I assume this is no longer necessary for newer versions of ubuntu.  Should we have some logic to exclude this PPA on the latest releases?

Comment (maxrothman) [inline playbooks/roles/common/tasks/main.yml:39]: I think in the future we'll still want to keep git current and Ubuntu's default package won't do that. Either way, I think that's out-of-scope for this change (I just moved the task around).

File: playbooks/roles/supervisor/tasks/main.yml [hunk 5]
```diff
@@ -121,6 +121,7 @@
     dest: "/etc/init/{{ supervisor_service }}.conf"
     owner: root
     group: root
+  when: ansible_distribution_major_version < 16
```
Comment (fredsmith) [inline playbooks/roles/supervisor/tasks/main.yml]: we should check for ansible_distribution = "Ubuntu"

**Comment by fredsmith** [comment]
I have concerns about how you're limiting tasks to Ubuntu >= 16 and how the git ppa is being installed even where not necessary but other than those, 👍

**Comment by maxrothman** [comment]
@fredsmith fixed. Rereview?

**Comment by fredsmith** [comment]
👍
